app/models/notification.rb

```diff
@@ -123,6 +126,30 @@ def target_status
     end
   end
 
+  def set_group_key!
+    return if filtered || Notification::GROUPABLE_NOTIFICATION_TYPES.exclude?(type)
```

Comment:
Nitpick, for style consistency:
```suggestion
    return if filtered? || Notification::GROUPABLE_NOTIFICATION_TYPES.exclude?(type)
```